tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -201,6 +216,84 @@ private int Serialize(ref byte[] bytes, int offset, in SpanModel spanModel)
                 offset += WriteMetaStruct(ref bytes, offset, in spanModel);
             }
 
+            if (hasSpanLinks)
+            {
+                offset += WriteSpanLink(ref bytes, offset, in spanModel);
+            }
+
+            return offset - originalOffset;
+        }
+
+        private int WriteSpanLink(ref byte[] bytes, int offset, in SpanModel spanModel)
+        {
+            int originalOffset = offset;
+            offset += MessagePackBinary.WriteStringBytes(ref bytes, offset, _spanLinkBytes);
+            offset += MessagePackBinary.WriteArrayHeader(ref bytes, offset, spanModel.Span.SpanLinks.Count);
+            foreach (var spanLink in spanModel.Span.SpanLinks)
+            {
+                var context = spanLink.Context;
+                var samplingPriority = context.TraceContext?.SamplingPriority ?? context.SamplingPriority;
+                var traceFlags = samplingPriority switch
+                {
+                    null => 0u,             // not set
+                    > 0 => 1u + (1u << 31), // keep
+                    <= 0 => 1u << 31,       // drop
+                };
+                var len = 3;
+                // check to serialize tracestate
+                if (context.IsRemote)
+                {
+                    len++;
+                }
+
+                if (traceFlags > 0)
```

Comment:
From my understanding, I think this should be checking whether `traceFlags is not null` as if it is `0` we wouldn't currently serialize the value and from the RFC it seems that the keep/drop values should be included and it should only be omitted if the sampling isn't known.

Copying this comment to the other location as well.

Reply:
There a `switch` above that sets traceFlags (not nullable):

- 0 means no value
- 1 + 31st bit means keep
- 0 + 31st bit means drop

This is specified in the RFC because some part of the system can't distinguish between `0` and `null`.

Reply:
Ahhhh okay I misread the RFC thanks @lucaspimentel

Reply:
I mean, I don't really understand why we didn't have 3 options:
- not set
- 0 for drop
- 1 for keep

But that's the RFC as approved 🤷🏽‍♂️